src/lib/BUTTON/devButton.cpp

```diff
@@ -4,93 +4,58 @@
 #include "common.h"
 #include "logging.h"
 #include "button.h"
-
-static Button button;
+#include "config.h"
+#include "devButton.h"
 
 #ifndef GPIO_BUTTON_INVERTED
 #define GPIO_BUTTON_INVERTED false
 #endif
+#ifndef GPIO_BUTTON2_INVERTED
+#define GPIO_BUTTON2_INVERTED false
+#endif
+#if !defined(GPIO_PIN_BUTTON2)
+#define GPIO_PIN_BUTTON2 UNDEF_PIN
+#endif
 
-#if defined(TARGET_TX)
-#include "POWERMGNT.h"
-void EnterBindingMode();
-
-static void enterBindMode3Click()
-{
-    if (button.getCount() == 3)
-    {
-        EnterBindingMode();
-    }
-};
-
-static void cyclePower()
-{
-    // Only change power if we are running normally
-    if (connectionState < MODE_STATES)
-    {
-        PowerLevels_e curr = POWERMGNT::currPower();
-        if (curr == POWERMGNT::getMaxPower())
-        {
-            POWERMGNT::setPower(POWERMGNT::getMinPower());
-        }
-        else
-        {
-            POWERMGNT::incPower();
-        }
-        devicesTriggerEvent();
-    }
-};
+static Button button1;
+#if defined(GPIO_PIN_BUTTON2)
+static Button button2;
 #endif
 
 #if defined(TARGET_RX)
-#if defined(PLATFORM_ESP32)
-#include <SPIFFS.h>
-#elif defined(PLATFORM_ESP8266)
-#include <FS.h>
+static constexpr struct {
+    bool pressType;
+    uint8_t count;
+    action_e action;
+} button_actions[2] = {
+    {true, 3, ACTION_START_WIFI},
+    {true, 7, ACTION_RESET_REBOOT}
+};
 #endif
 
-extern void setWifiUpdateMode();
+static ButtonAction_fn actions[ACTION_LAST] = { nullptr };
 
-static void buttonRxLong()
+void registerButtonFunction(action_e action, ButtonAction_fn function)
 {
-#if defined(PLATFORM_ESP32) || defined(PLATFORM_ESP8266)
-    // ESP/ESP32 goes to wifi mode in 5x longpress
-    if (button.getLongCount() > 4 && connectionState != wifiUpdate)
-    {
-        setWifiUpdateMode();
-    }
-#endif
-
-    // All RX reset their config in 9x longpress and reboot
-    if (button.getLongCount() > 8)
-    {
-        config.SetDefaults(true);
-#if defined(PLATFORM_ESP32) || defined(PLATFORM_ESP8266)
-        // Prevent WDT from rebooting too early if
-        // all this flash write is taking too long
-        yield();
-        // Remove options.json and hardware.json
-        SPIFFS.format();
-        ESP.restart();
-#elif defined(PLATFORM_STM32)
-        HAL_NVIC_SystemReset();
-#endif
-    }
+    actions[action] = function;
 }
-#endif
 
-static void initialize()
+static void handlePress(uint8_t button, bool longPress, uint8_t count)
 {
-    if (GPIO_PIN_BUTTON != UNDEF_PIN)
+    std::list<action_t>::iterator it;
+    DBGLN("handle press");
```

Comment:
`DBGLN("handlePress(%u,%u,%u)", button, (uint8_t)longPress, count);` so this conveys some information, and switch the one in button.cpp to a `DBGVLN` so the same information isn't doubled up